data/api/client-server/room_event_by_timestamp.yaml

```diff
@@ -0,0 +1,108 @@
+# Copyright 2022 The Matrix.org Foundation C.I.C.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+swagger: '2.0'
+info:
+  title: "Matrix Client-Server Rooms by date API"
+  version: "1.0.0"
+host: localhost:8008
+schemes:
+  - https
+  - http
+basePath: /_matrix/client/v1
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  $ref: definitions/security.yaml
+paths:
+  "/rooms/{roomId}/timestamp_to_event":
+    get:
+      x-addedInMatrixVersion: "1.6"
+      summary: Get the closest event ID to the given timestamp
```

Comment:
I'm not sure about the syntax in these yaml files for where best to add it but seems good to add a section about ["Receiving an unrenderable event ID"](https://github.com/matrix-org/matrix-spec-proposals/pull/3030/files#diff-64e5c5a583fa79fac1ed7856586bd15b841379573dde1a9839d1e3c4c8c370cfR116) with the text from the MSC. Just some tips on how to handle the edge case.

Reply:
I added a paragraph about that, but I tried to make it more generic-sounding than what's in the MSC.  Let me know if I've butchered it too much.

Reply:
Looks good @uhoreg Good addition 👍